plugin.py

```diff
@@ -820,6 +840,14 @@ def updateDeviceSettings(self, devicename, devicetype, config):
                 oldconfigdict = json.loads(device.Options['config'])
             except (ValueError, KeyError, TypeError) as e:
                 pass
+            # Sensor support
+            # Correction for subsequent messages
```

Comment:
Can you clarify this comment a bit, what is corrected?

Reply:
BLE sensors are reported by HA discovery each line per sensor capability, like below. Current plugin design is to parse each line individually. In example below, a temperature sensor will be detected. Then in next line the temperature sensor will be switched to humidity only sensor. The code in question checks for this situation and only allows to temperature only sensor to upgrade to temp+hum sensor, but not to downgrade to humidity only. While doing it, I want to keep original dictionary from the 1st temperature line, which is leading

`11:03:44.772 MQT: homeassistant/sensor/LivingRoom/Temperature/config = {"availability":[],"device":{"identifiers":["BLELivingRoom"],"name":"LivingRoom","manufacturer":"tas","model":"ATC","via_device":"ESP32-FFHeatingActuators_eth"},"dev_cla":"temperature","expire_after":600,"json_attr_t":"tele/tasmota_ble/LivingRoom","name":"LivingRoom_Temperature","state_topic":"tele/tasmota_ble/LivingRoom","uniq_id":"LivingRoom_Temperature","unit_of_meas":"°C","val_tpl":"{{ value_json.Temperature }}"}
11:03:44.815 MQT: homeassistant/sensor/LivingRoom/Humidity/config = {"availability":[],"device":{"identifiers":["BLELivingRoom"],"name":"LivingRoom","manufacturer":"tas","model":"ATC","via_device":"ESP32-FFHeatingActuators_eth"},"dev_cla":"humidity","expire_after":600,"json_attr_t":"tele/tasmota_ble/LivingRoom","name":"LivingRoom_Humidity","state_topic":"tele/tasmota_ble/LivingRoom","uniq_id":"LivingRoom_Humidity","unit_of_meas":"%","val_tpl":"{{ value_json.Humidity }}"}
11:03:44.854 MQT: homeassistant/sensor/LivingRoom/DewPoint/config = {"availability":[],"device":{"identifiers":["BLELivingRoom"],"name":"LivingRoom","manufacturer":"tas","model":"ATC","via_device":"ESP32-FFHeatingActuators_eth"},"dev_cla":"temperature","expire_after":600,"json_attr_t":"tele/tasmota_ble/LivingRoom","name":"LivingRoom_DewPoint","state_topic":"tele/tasmota_ble/LivingRoom","uniq_id":"LivingRoom_DewPoint","unit_of_meas":"°C","val_tpl":"{{ value_json.DewPoint }}"}
11:03:44.895 MQT: homeassistant/sensor/LivingRoom/Battery/config = {"availability":[],"device":{"identifiers":["BLELivingRoom"],"name":"LivingRoom","manufacturer":"tas","model":"ATC","via_device":"ESP32-FFHeatingActuators_eth"},"dev_cla":"battery","expire_after":600,"json_attr_t":"tele/tasmota_ble/LivingRoom","name":"LivingRoom_Battery","state_topic":"tele/tasmota_ble/LivingRoom","uniq_id":"LivingRoom_Battery","unit_of_meas":"%","val_tpl":"{{ value_json.Battery }}"}
11:03:44.936 MQT: homeassistant/sensor/LivingRoom/RSSI/config = {"availability":[],"device":{"identifiers":["BLELivingRoom"],"name":"LivingRoom","manufacturer":"tas","model":"ATC","via_device":"ESP32-FFHeatingActuators_eth"},"dev_cla":"signal_strength","expire_after":600,"json_attr_t":"tele/tasmota_ble/LivingRoom","name":"LivingRoom_RSSI","state_topic":"tele/tasmota_ble/LivingRoom","uniq_id":"LivingRoom_RSSI","unit_of_meas":"dB","val_tpl":"{{ value_json.RSSI }}"}
11:03:44.980 MQT: tele/tasmota_ble/LivingRoom = {"Time":"2021-12-16T11:03:44","alias":"LivingRoom","mac":"a4c138334b8e","Temperature":21.5,"Humidity":58.0,"DewPoint":12.9,"Battery":80,"RSSI":-95}`